stable/database/files/nuosm

```diff
@@ -402,64 +402,74 @@ function checkBackupId() {
 function loadFromSnapshot() {
   local recreate_archives="false"
 
-  if [ ! -f "$DB_DIR/state.dat" ]; then
-    local archives=$( find "/var/opt/nuodb/archive" -name info.json )
-    if [ -z "$archives" ] || [ $(echo "$archives" | wc -l) != 1 ]; then 
+  if [ ! -f "$DB_DIR/info.json" ]; then
+    local archives="$(find /var/opt/nuodb/archive -name info.json)"
+    if [ -z "$archives" ] || [ "$(echo "$archives" | wc -l)" != 1 ]; then
+      # Relax check for archive snapshot for SMs other than ordinal 0. It is
+      # possible that a database that was restored from a backup with
+      # pre-provisioned PVCs has been scaled up. In this case, the new SMs will
+      # have empty PVCs, so no archive snapshot will be found.
+      local ordinal="${POD_NAME##*-}"
+      if [ -z "$archives" ] && [ "$ordinal" != 0 ]; then
+        log "No archive snapshot found for $POD_NAME with non-0 ordinal $ordinal"
+        return 1
+      fi
       die 1 "Did not find exactly 1 archive: ${archives}"
     fi
 
-    local archive_path=$( dirname $archives )
+    local archive_path="$(dirname "$archives")"
     log "Found snapshot at ${archive_path}"
 
     if [ -n "$BACKUP_ID" ]; then
```

Comment:
That sounds good, but we have to remember to check that the in-place restore is not being performed with such a backup.

This would generate a matrix of supported configurations:

| | cold snapshot | hot snapshot |
|---|---|---|
| in-place | unsupported | supported |
| clone | supported | supported |

I'll leave this change for a different PR.

Reply:
This is mainly to support executing backups without calling backup hooks. Such backups won't support `in-place` restore as you've outlined.